test/unit/htmlValidatorTest.js

```diff
@@ -0,0 +1,508 @@
+        }
+
+        // check to see tha tthe page did not validate
+        let test1 = res.text.includes('HTML did not pass validation')
+        assert.equal(test1, true)
+        testApp.kill('SIGINT')
+      })
+      testApp.on('exit', () => {
+        done()
+      })
+    })
+  })
+
+  it('should be able to run the validator even if we change the port number of the validator', function (done) {
+    // generate the app
+    generateTestApp({
+      generateFolderStructure: true,
+      appDir: appDir,
+      htmlValidator: {
+        enable: true,
+        suppressWarnings: true,
+        port: 3000
+      },
+      onServerStart: true
+    }, options)
+
+    // fork the app and run it as a child process
+    const testApp = fork(path.join(appDir, 'app.js'), ['--dev'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // on the server coming to life, see if we can send a request to the new port
+    testApp.on('message', (params) => {
+      request(`http://localhost:${params.htmlValidator.port}`)
+      .get('/')
+      .expect(200, (err, res) => {
+        if (err) {
+          assert.fail(err)
+          testApp.kill('SIGINT')
+        }
+
+        // check to see that the page loaded
+        let test1 = res.status
+        assert.equal(test1, 200)
+        testApp.kill('SIGINT')
+      })
+      testApp.on('exit', () => {
+        done()
+      })
+    })
+  })
+
+  it('should be able to run the htmlValidator independently from the app if the seperateProcess param is set to true', function (done) {
+    // generate the app
+    generateTestApp({
+      generateFolderStructure: true,
+      appDir: appDir,
+      htmlValidator: {
+        enable: true,
+        suppressWarnings: true,
+        separateProcess: true,
+        port: 8888
+      },
+      onServerStart: true
+    }, options)
+
+    // fork the app and run it as a child process
+    const testApp = fork(path.join(appDir, 'app.js'), ['--dev'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // test to see that the validator still works
+    testApp.on('message', (params) => {
+      // variable to check if we had an error on the test app
+      let testAppError = false
+      request(`http://localhost:${params.port}`)
+      .get('/Broken')
+      .expect(200, (err, res) => {
+        if (err) {
+          assert.fail(err)
+          testApp.kill('SIGINT')
+          testAppError = true
+        }
+        // check to see that the page did validate and failed
+        let test1 = res.text.includes('HTML did not pass validation')
+        assert.equal(test1, true)
+        testApp.kill('SIGINT')
+      })
+
+      // check to see if the validator is accessible (should get back 200)
+      request(`http://localhost:${params.htmlValidator.port}`)
+      .get('/')
+      .expect(200, (err, res) => {
+        if (err) {
+          assert.fail(err)
+          const killLine = fork(path.join(__dirname, '../', '../', 'lib', 'scripts', 'killValidator.js'))
```

Comment:
You don't have to worry about path.join when using this fork api. Also if you pass the same params to `killValidator` that you pass to the testApp fork there won't be any logging.

Reply:
fix the merge conflict too